src/CMakeLists.txt

```diff
@@ -79,6 +90,14 @@ if(ENABLE_DEADLINE_MISSED)
   add_definitions(-DDDSI_INCLUDE_DEADLINE_MISSED)
 endif()
 
+if(ENABLE_TYPE_DISCOVERY)
+  add_definitions(-DDDSI_INCLUDE_TYPE_DISCOVERY)
+endif()
+
+if(ENABLE_TOPIC_DISCOVERY)
+  add_definitions(-DDDSI_INCLUDE_TOPIC_DISCOVERY)
+endif()
+
```

Comment:
I agree that it would be better to narrow the scope for these definitions, but downside is that the definitions cannot be used outside ddsc/ddsi. Currently the `ddsls` tool also uses it for including (or not) DCPSTopic support, so we'll also need to define it for that target (which cannot be done in /src/CMakeLists.txt, because the targets are not yet created at that point).